Drupal/modules/custom/weather/src/Plugin/Block/WeatherDisplay.php

```diff
@@ -0,0 +1,174 @@
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->config = $config_factory;
+    $this->httpClient = $http_client;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+      $container->get('http_client'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [
+      'city' => '',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockForm($form, FormStateInterface $form_state) {
+    $form['city'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('City'),
+      '#description' => $this->t('Please enter the city to show weather details. Eg: Mumbai, India'),
+      '#default_value' => $this->configuration['city'],
+      '#required' => TRUE,
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockSubmit($form, FormStateInterface $form_state) {
+    $this->configuration['city'] = $form_state->getValue('city');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    // Get weather settings.
+    $weather_settings = $this->config->get(WeatherSettingsForm::SETTINGS);
+
+    // Bail out if the config is empty.
+    if (empty($weather_settings)) {
+      return [];
+    }
+
+    // Get location details from session.
+    $city = $this->configuration['city'];
+    if (!$city) {
+      $city_not_found_message = $this->t('No city entered. Please enter a city to see the weather details');
+      return [
+        '#markup' => '<div class="weather--error">' . $city_not_found_message . '</div>',
+      ];
+    }
+
+    // Endpoint that should be hit.
+    $request_url = $weather_settings->get('base_url') . '/current.json';
+    // Make a HTTP GET request to the endpoint.
+    try {
+      $request = $this->httpClient->request('GET', $request_url, [
+        'query' => [
+          'key' => $weather_settings->get('api_key'),
+          'q' => $city,
+          'units' => 'metric',
+        ],
+      ]);
+
+      // Parse the response.
+      $response = Json::decode($request->getBody());
+      $weather_data = [];
+      $weather_data['location'] = $response['location']['name'] . ', ' . $response['location']['region'] . ', ' . $response['location']['country'];
+      $weather_data['temperature'] = $response['current']['temp_c'];
+      $weather_data['feels_like'] = $response['current']['feelslike_c'];
+      $weather_data['weather_condition_icon'] = $response['current']['condition']['icon'];
+      $weather_data['weather_condition_text'] = $response['current']['condition']['text'];
+      $weather_data['wind'] = $response['current']['wind_kph'];
+      $weather_data['precipitation'] = $response['current']['precip_mm'];
+
```

Comment:
@iyyappan-axelerant I was thinking of having a service file initially then dropped the idea considering it would be used only in this location. But, yes from a wider POV, it makes sense to have this as a service. I can add a service file!

Thank you for reviewing the PR and providing feedback!